src/components/chart/state-history-chart-line.ts

```diff
@@ -256,7 +256,21 @@ export class StateHistoryChartLine extends LitElement {
           axisLabel: {
             margin: 5,
             formatter: (value: number) => {
-              const label = formatNumber(value, this.hass.locale);
+              const formatOptions =
+                value >= 1 || value <= -1
+                  ? undefined
+                  : {
+                      // show the first significant digit for tiny values
+                      maximumFractionDigits: Math.max(
+                        2,
+                        -Math.floor(Math.log10(Math.abs(value % 1 || 1)))
+                      ),
+                    };
+              const label = formatNumber(
+                value,
+                this.hass.locale,
+                formatOptions
+              );
```

Comment:
Why don't we use the display precision from the entity registry for this one?

Reply:
There could be many entities. Even if it is only 1, it doesn't make sense to show axis labels that are all the same like is shown in the issue